src/main/java/managers/SaveManager.java

```diff
@@ -0,0 +1,109 @@
+package managers;
+
+import errors.InvalidDeadlineException;
+import errors.InvalidEventException;
+import tasks.Deadlines;
+import tasks.Events;
+import tasks.Task;
+import tasks.ToDos;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class SaveManager {
+
+    public static final String PATHNAME = "data.txt";
+
+    //Storing format:
+    //T0 name
+    //D0 name /by time
+    //E0 name /from time /to time
+    public static ArrayList<Task> initialiseData() {
+        File f = new File(PATHNAME);
+        ArrayList<Task> storedTasks = new ArrayList<>();
+        try {
+            if (f.createNewFile()) {
+                return storedTasks;
+            }
+        } catch (IOException e) {
+            System.out.println("Error occurred when creating new file.");
+        }
+        try {
+            Scanner scanner = new Scanner(f);
+            while (scanner.hasNext()) {
+                String line = scanner.nextLine();
+                processStoredTask(storedTasks, line);
+            }
+        } catch (FileNotFoundException e) {
+            return storedTasks;
+        } catch (InvalidDeadlineException | InvalidEventException e) {
+            System.out.println("Stored data is corrupted.");
+        }
+
+        return storedTasks;
+    }
+
+    public static void saveCurrentState (ArrayList<Task> todoList) {
+        try {
+            FileWriter fileWriter = new FileWriter(PATHNAME, true);
+            FileWriter fileClearer = new FileWriter(PATHNAME);
+            fileClearer.write("");
+            fileClearer.close();
+            String taskState = "";
+            for (Task item: todoList) {
+                String toWrite = "";
+                if (item.isDone()) {
+                    taskState = "1";
+                } else {
+                    taskState = "0";
+                }
+                switch (item.getClassType()) {
+                case "T":
+                    toWrite = "T" + taskState + item.getToStore();
+                    break;
+                case "D" :
+                    toWrite = "D" + taskState + item.getToStore();
+                    break;
+                case "E":
+                    toWrite = "E" + taskState + item.getToStore();
+                    break;
```

Comment:
It's generally a good practice to include the default branch of switch case. You could use it to throw an error in this case.

In fact, is the switch case really needed here? You might want to consider using polymorphism to make this piece of code scalable (consider the case where other types of tasks will be added to your program, how do you change this code to cater such changes so that you don't have to update this piece of code every time a new type of task is added?).